x-pack/plugin/ilm/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/indexlifecycle/CCRIndexLifecycleIT.java

```diff
@@ -368,6 +326,70 @@ public void testUnfollowInjectedBeforeShrink() throws Exception {
         }
     }
 
+    public void testCannotShrinkLeaderIndex() throws Exception {
+        String indexName = "shrink-leader-test";
```

Comment:
I chatted about this test @gwbrown. The stopping of ilm in this test can be problematic for other tests that run after it. First all tests are executed as part of the leader cluster test task and then as part of the follower cluster test task (this while leader test cluster is running). The other which the test methods are executed is randomized by the test framework. 

In order to avoid this problem the entire test can be ran just against the follow cluster test task, like is done below.
Also what also is possible is to remove the stop and start ilm calls and in instead setup the leader index first in leader cluster, execute put follow in follower cluster and then create the leader ilm policy. 

```java
if ("leader".equals(targetCluster)) {
            return;
        } else if ("follow".equals(targetCluster) == false) {
            throw  new AssertionError("unexpected target cluster [" + targetCluster + "]");
        }

        String indexName = "shrink-leader-test";
        String shrunkenIndexName = "shrink-" + indexName;
        String policyName = "shrink-leader-test-policy";
        try (RestClient leaderClient = buildLeaderClient()) {

            putShrinkOnlyPolicy(leaderClient, policyName);

            // Stop ILM to let us set up the follower before the policy swings into action and shrinks the index
            stopILM(client());
            assertBusy(() -> assertEquals("STOPPED", ilmStatus(client())));
            Settings indexSettings = Settings.builder()
                .put("index.soft_deletes.enabled", true)
                .put("index.number_of_shards", 2)
                .put("index.number_of_replicas", 0)
                .put("index.lifecycle.name", policyName)
                .build();
            createIndex(indexName, indexSettings, "", "", leaderClient); // <-- method does not yet exist
            ensureGreen(indexName, leaderClient); // <-- method does not yet exist

            // Policy with the same name must exist in follower cluster too:
            putUnfollowOnlyPolicy(client(), policyName);
            followIndex(indexName, indexName);
            ensureGreen(indexName);

            startILM(leaderClient);
            index(leaderClient, indexName, "1");
            assertDocumentExists(leaderClient, indexName, "1");

            assertBusy(() -> {
                assertDocumentExists(client(), indexName, "1");
                // Sanity check that following_index setting has been set, so that we can verify later that this setting has been unset:
                assertThat(getIndexSetting(client(), indexName, "index.xpack.ccr.following_index"), equalTo("true"));

                // We should get into a state with these policies where both leader and followers are waiting on each other
                assertILMPolicy(leaderClient, indexName, policyName, "warm", "shrink", "wait-for-shard-history-leases");
                assertILMPolicy(client(), indexName, policyName, "hot", "unfollow", "wait-for-indexing-complete");
            });

            // Manually set this to kick the process
            updateIndexSettings(leaderClient, indexName, Settings.builder()
                .put("index.lifecycle.indexing_complete", true)
                .build()
            );

            assertBusy(() -> {
                // The shrunken index should now be created on the leader...
                Response shrunkenIndexExistsResponse = leaderClient.performRequest(new Request("HEAD", "/" + shrunkenIndexName));
                assertEquals(RestStatus.OK.getStatus(), shrunkenIndexExistsResponse.getStatusLine().getStatusCode());

                // And both of these should now finish their policies
                assertILMPolicy(leaderClient, shrunkenIndexName, policyName, "completed");
                assertILMPolicy(client(), indexName, policyName, "completed");
            });
        }
```